gamemode/modules/police/sv_init.lua

```diff
@@ -358,7 +358,7 @@ function DarkRP.hooks:playerArrested(ply, time, arrester)
     umsg.End()
 end
 
-function DarkRP.hooks:playerUnArrested(ply, actor)
+function DarkRP.hooks:playerUnArrested(ply, actor, teleportOverride)
```

Comment:
This hook does quite a few different things that someone might want to override:

- Exiting a vehicle
- Waking up from sleep
- Setting  respawn time
- Give the player weapons through `PlayerLoadout`
- Teleport to a spawn position if the config is enabled

Your use case is the last option, but I can imagine future PRs adding more arguments to deal with the other behaviours. The same can then be done for the other hooks. It's a bit of a slippery slope, but I think it is a good indication that this might not be the best way to go here.

Instead, would it be an idea to override `DarkRP.hooks.playerUnArrested` to remove the behaviours you don't want? Then in a separate hook (or in the override), you can have your own teleporting implementation.

Reply:
I see your point. I do think the respawning aspect is probably the most important as I had literally no clue those other things happened without reading the code. I've seen many different addons like jail escaping systems, bail npcs, etc. and all of them seem to have subpar workarounds for unarresting a player without respawning them. I suppose I could just keep this change for my server only, but I think it's something a lot of other people could benefit from. The problem is that I do want players to be respawned when they get unarrested, but in specific cases I want to be able to choose. For example, if a player breaks another player out I don't want the escaping player to be respawned.